### ISSUE TITLE
Confusing terminology around "Custom Collectors"

### ISSUE THREAD

**Issue by directionless** [discourse/prometheus_exporter#28]
I'm confused about what "Custom Collectors" are. In other prometheus libraries, they're a way to proxy metrics. For example [python](https://github.com/prometheus/client_python/#custom-collectors) and [java](https://github.com/prometheus/client_java#custom-collectors).

The use here does not seem to match. Instead, this seems to be about setting a batch of messages. But still working with them in the same `increment` way. 

Am I misunderstanding, or is this fundamentally different?

**Comment by SamSaffron** [comment]
Yes the idea behind custom collector is that they allow the master exporter process to be less "chattery" it allows us to send a batch of metrics more efficiently from the processes being monitored vs the very chattery single message for every metric. 

The collector can do whatever it wants with the data it gets from the processes it is collecting for, including increment, summary and so on.

**Comment by directionless** [comment]
That sounds quite different than the prometheus use of the term. Is it possible to change terminology?

I'm also interesting in the prometheus functionality. As an example, in my rails apps I want to export both http traffic metrics (which requires some pre-fork support) and also some straight rails metrics (like activerecord object counts). I'm not sure I can easily do that in this framework.
